images/base/Dockerfile

```diff
@@ -55,8 +55,22 @@ ARG CNI_PLUGINS_AMD64_SHA256SUM="962100bbc4baeaaa5748cdbfce941f756b1531c2eadb290
 ARG CNI_PLUGINS_ARM64_SHA256SUM="ef17764ffd6cdcb16d76401bac1db6acc050c9b088f1be5efa0e094ea3b01df0"
 ARG CNI_PLUGINS_PPC64LE_SHA256SUM="5bd3c82ef248e5c6cc388f25545aa5a7d318778e5f9bc0a31475361bb27acefe"
 
-# copy in static files (configs, scripts)
-COPY files/ /
+# copy in static files
+# all scripts are 0755: http://www.filepermissions.com/file-permission/0755
+COPY --chmod=0755 files/usr/local/bin/* /usr/local/bin/
+
+# all directories are 0755: http://www.filepermissions.com/file-permission/0755
+COPY --chmod=0755 files/etc/containerd  files/etc/default files/etc/sysctl.d files/etc/systemd  /etc/
+COPY --chmod=0755 files/etc/systemd/system  /etc/systemd/
+COPY --chmod=0755 files/etc/systemd/system/kubelet.service.d  /etc/systemd/system/
+
+# all configs are 0644: http://www.filepermissions.com/file-permission/0644
+COPY --chmod=0644 files/etc/* /etc/
+COPY --chmod=0644 files/etc/containerd/* /etc/containerd/
+COPY --chmod=0644 files/etc/default/* /etc/default/
+COPY --chmod=0644 files/etc/sysctl.d/* /etc/sysctl.d/
+COPY --chmod=0644 files/etc/systemd/system/* /etc/systemd/system/
+COPY --chmod=0644 files/etc/systemd/system/kubelet.service.d/* /etc/systemd/system/kubelet.service.d/
```

Comment:
this wound up really verbose.
maybe we should script creating a tarball in the makefile and then ADD it :/
at least then we could maybe programattically say "directories are X, files are Y, scripts are Z"

Reply:
:)

Reply:
there's some issues that approach too. discussed offline wtih amit, with correct usage of `COPY` (non recursive) directories will default to 755 like we want and it's not more verbose then.

Reply:
know I was wrong because I was assuming tar from a host, not from files in a git repo, but was surprisingly nice to see you reconsider it XD